server/controllers/RatingController.js

```diff
@@ -0,0 +1,215 @@
+import sequelize from 'sequelize';
+import models from '../database/models';
+import articleRating from '../utils/services/articleRating';
+
+const { Article, Rating } = models;
+class RatingController {
+  /**
+   * post the rating of the article
+   * with slug given in req.params
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @returns {Object} res.body contain the info of the rating
+   * res.body is object with properties ratingDetail and result
+   */
+  static rateArticle(req, res) {
+    let { rating } = req.body;
+    let userId = req.body.decoded.id;
+    const { slug } = req.params;
+    rating = parseInt(rating, 10);
+    userId = parseInt(userId, 10);
+    return Article
+      .find({
+        where: {
+          slug,
+        },
+      })
+      .then((article) => {
+        if (article) {
+          if (article.authorId === parseInt(userId, 10)) {
+            return res.status(403).json({
+              status: 'fail',
+              message: 'You can\'t provide ratings for your article',
+            });
+          }
+          return Rating
+            .find({
+              where: {
+                userId,
+                articleId: article.id,
+              },
+            })
+            .then((rate) => {
+              if (!rate) {
+                return Rating
+                  .create({
+                    rating,
+                    articleId: article.id,
+                    userId,
+                  })
+                  .then(rated => Rating
```

Comment:
You can reduce the lines of code here by using the sequelize `findOrCreate` method instead of finding before creating. Another approach (if the relationship is done well) is to use something like `article.createRating`